### DISCUSSION TITLE
Simple Logic plugin

### DISCUSSION THREAD

**Discussion by mavencode01** [Kong/kong#7390] [Ideas and feature requests]
One really good thing about Kong is the plugin architecture it has and I think it would be nice able to add `simple logic plugin` to check the user/consumer status without delegating this task to upstream services.

For example, Account Confirmation plugin will check every request that the user / consumer status is `verified`

**Comment by Tieske** [comment]
Can you provide a more detailed description of your use case?

**Comment by mavencode01** [comment]
Sure, I'm building a new system where I have to ensure every users/consumers of my microservices has verified his account during registration.   

Am thinking if there is a simple plugin for this, I don't have to do an extra check in all my services.

**Comment by Tieske** [comment]
Thx. How would you envision this to work? What interaction between the application and kong would be required? Can you describe this flow?

**Comment by mavencode01** [comment]
I haven't looked into how the plugin works but on a high level, I would imagine when you create a consumer in Kong, you can specify extra fields just like you would for a normal user (verified, dob, etc. ) as json.  
After that, you  add the `SimpleLogic` plugin that adds one or two multiple conditions.

```
user.verified = true
user.age > 18
...

```

On each request, the plugin evaluates the conditions and can deny or allow the request.
